src/rgw/rgw_op.cc

```diff
@@ -3388,6 +3536,24 @@ void RGWDeleteObj::execute()
       if (op_ret >= 0) {
         delete_marker = del_op.result.delete_marker;
         version_id = del_op.result.version_id;
+        map<string, bufferlist>::iterator cmp = s->bucket_attrs.find(RGW_ATTR_COMPRESSION);
+        if (cmp != s->bucket_attrs.end()) {
+          uint64_t bucket_size, deleted_size;
+          ::decode(bucket_size, cmp->second);
+          bool tmp;
+          RGWCompressionInfo cs_info;
+          rgw_compression_info_from_attrset(attrs, tmp, cs_info);
```

Comment:
if this returns an error, `tmp` won't be initialized. need to either check the return value here, or initialize `bool tmp = false`. if we do ignore the return value, please document that with `(void) rgw_compression_info_from_attrset(...)`